lib/auth/auth.go

```diff
@@ -5896,13 +5901,20 @@ func (a *Server) validateMFAAuthResponseForRegister(
 	return true, nil
 }
 
-// ValidateMFAAuthResponse validates an MFA or passwordless challenge.
-// Returns the device used to solve the challenge (if applicable) and the
-// username.
-func (a *Server) ValidateMFAAuthResponse(ctx context.Context, resp *proto.MFAAuthenticateResponse, user string, passwordless bool) (*types.MFADevice, string, error) {
+// ValidateMFAAuthResponse validates an MFA or passwordless challenge. If provided,
+// required challenge extensions will be checked against the stored challenge when
```

Comment:
The wording makes it sound like `requiredExtensions` are optional, but there's a return guard disallowing this.